markdown_lab/network/async_client.py

```
@@ -0,0 +1,533 @@
+        
+        Args:
+            url: URL to fetch
+            **kwargs: Additional request arguments
+            
+        Returns:
+            Tuple of (url, content_or_exception)
+        """
+        async with self._semaphore:
+            try:
+                # Add per-domain rate limiting if configured
+                domain = urlparse(url).netloc
```

Ruff (0.11.9)

Local variable `domain` is assigned to but never used

Remove assignment to unused variable `domain`

(F841)